app/models/miq_request_workflow.rb

```diff
@@ -840,12 +840,7 @@ def allowed_ci(ci, relats, sources, filtered_ids = nil)
     result.each_with_object({}) { |s, hash| hash[s[0]] = s[1] }
   end
 
-  def process_filter(filter_prop, ci_klass, targets = [])
-    return targets if targets.blank?
```

Comment:
:+1:  This empty array check is already handled in the Rbac.filtered and MiqSearch calls so it is not need here.